lib/mediawiki/butt.rb

```diff
@@ -50,12 +50,19 @@ def post(params, autoparse = true, header = nil)
       header['User-Agent'] = @logged_in ? "#{@name}/MediaWiki::Butt" : 'NotLoggedIn/MediaWiki::Butt'
       header['User-Agent'] = @custom_agent if defined? @custom_agent
 
-      res = @client.post(@uri, params, header)
-
-      if autoparse
-        return JSON.parse(res.body)
-      else
-        return res
+      @retauto = {}
+      @retstr = ''
+      @lastContinue = {"continue": ""}
+      while true:
```

Comment:
I dunno how to do half the stuff in ruby, I do python bro. What's wrong with it, go fix it and why does travis say it werks en xD